toutv/bos.py

```diff
@@ -520,13 +521,24 @@ def _get_playlist_url(self):
         params = dict(toutv.config.TOUTV_PLAYLIST_PARAMS)
         params['idMedia'] = self.PID
 
-        r = self._do_request(url, params=params)
-        response_obj = r.json()
+        num_tries = 0
 
-        if response_obj['errorCode']:
-            raise RuntimeError(response_obj['message'])
+        def finished():
+            return num_tries == 3
 
-        return response_obj['url']
+        while not finished():
+            r = self._do_request(url, params=params)
+            try:
+                response_obj = r.json()
+                if response_obj['errorCode']:
+                    raise RuntimeError(response_obj['message'])
+                return response_obj['url']
+            except ValueError:
```

Comment:
I noticed your JSONDecodeError was different from mine. I would guess it depends on the JSON library used, which can differ between systems.